system/security/shared/caching.cpp

```diff
@@ -697,35 +721,38 @@ SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
 
     SecAccessFlags defaultPermission = SecAccess_None;
     const std::string username(user.getName());
-    bool addedToCache = false;
+
     {
-        CriticalBlock defaultScopePermissionBlock(syncDefaultScopePermissions);
+        ReadLockBlock readLock(m_defaultFilePermissionRWLock);
         auto it = m_userDefaultFileScopePermissions.find(username);
+        readLock.clear();   // lock no longer needed
```

Comment:
I think the lock is still needed - otherwise the comparison against end is invalid, also iter->second is invalid